### DISCUSSION TITLE
Translations FileLoader implementation work incorrectly with custom paths binding

### DISCUSSION THREAD

**Discussion by aprokopenko** [laravel/framework#49573] [Ideas]
### Laravel Version

10.39

### PHP Version

8.2.13

### Database Driver & Version

not related

### Description

In case you re-register translation Loader contract and add more paths, that the standard ones (which are illuminate langs and app langs) - then namespaces overrides works incorrectly.

The last element overrides all previous paths.

My vision that is should replace recursively all the available overrides, in this case you can specify only custom lines you need to change.

(In my case, I'm creating a package, which may override some vendor translations, but app should have higher priority)

### Steps To Reproduce

1. Register custom translation.loader binding:

```php
class TranslationServiceProvider extends IlluminateTranslationServiceProvider
{
    protected function registerLoader()
    {
        $this->app->singleton('translation.loader', function ($app) {
            $class = config('i18n.translation_loader', FileLoader::class);

            return new $class($app['files'], [
                base_path() . '/vendor/laravel/framework/src/Illuminate/Translation/lang',
                // we will add one more line in the middle as example of override paths
                base_path() . '/lib/override-package/lang',
                $app['path.lang'],
            ]);
        });
    }
}
```

2. Register this provider in config

3. Define some name lang namespace path in AppServiceProvider like 
$this->loadTranslationsFrom(base_path('lib/custom-package/lang'), 'package');

4. Create lang files in both lang paths:
```php
# lib/custom-package/lang/en/test.php

return [
 'foo' => 'bar',
];
```

```php
# lib/override-package/lang/en/test.php

return [
 'foo' => 'bar2',
 'fruit' => 'orange',
];
```

5. Try to get translation like `trans('package::test.foo')`. You will get `bar`, not `bar2`. Futhermore, you `trans('package::test.fruit')` also won't have a translation

**Comment by aprokopenko** [comment]
Would be great if this can be release as hotfix soon, because it blocks our work a bit :)

And thanks for the great work and that these parameters are their, event with small bug :)

**Comment by crynobone** [comment]
> My vision that is should replace recursively all the available overrides, in this case you can specify only custom lines you need to change.

This clearly indicates that this is a feature request and not a bug report.

**Comment by aprokopenko** [comment]
In any way, I added a Pull Request with all required unit tests, which cover the described situation:
https://github.com/laravel/framework/pull/49579
